### ISSUE TITLE
Re-assigning import modules is flagged as "not assignable"

### ISSUE THREAD

**Issue by nedbat** [astral-sh/ty#2690]
### Summary

This code:
```python
import os
from types import ModuleType

def mod_mod(mod: ModuleType) -> ModuleType:
    return mod

os = mod_mod(os)
```
produces:
```
% uvx ty check --python-version=3.14 /tmp/modassign.py
error[invalid-assignment]: Object of type `ModuleType` is not assignable to `<module 'os'>`
 --> /tmp/modassign.py:7:1
  |
5 |     return mod
6 |
7 | os = mod_mod(os)
  | --   ^^^^^^^^^^^ Incompatible value of type `ModuleType`
  | |
  | Declared type `<module 'os'>`
  |
info: rule `invalid-assignment` is enabled by default

Found 1 diagnostic
```

### Version

ty 0.0.14 (16597f5e2 2026-01-26)

**Comment by carljm** [comment]
Thanks for the report! The root cause here is that (unlike other type checkers) we currently consider an import to declare a type for a symbol (meaning you can't reassign it to a different or wider type without re-declaring it), and we probably shouldn't do that. See #1836.

**Comment by carljm** [comment]
(This also means that the workaround to make ty happy with this code is to explicitly re-declare `os` with `os: ModuleType = ...`: https://play.ty.dev/09b2a300-f7f5-4236-87c3-427d4e724224 -- because, also unlike other type checkers, we allow re-declaring or shadowing the same symbol with a new type.)
